### PR TITLE
Ensure that every script can access Redis via Unix socket

### PR DESCRIPTION
**Pull request by mbyczkowski**
## Pull Request Checklist

**Is this in reference to an existing issue?**

No, I think I should have created on, though.

#### General

- [x] Update Changelog following the conventions laid out on [Our CHANGELOG Guidelines ](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [X] RuboCop passes

- [X] Existing tests pass

#### New Plugins

- [x] Tests

- [x] Add the plugin to the README

- [x] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose

Fix a bug + improve reliability/maintainability of the scripts

#### Known Compatibility Issues

**TL;DR I believe I have introduced a bug in #32 that made _check-redis-ping.rb_ the only script that actually works with Unix sockets. This PR is a proper fix for this.**

I wanted to do a simple fix, but then realized that for the sake of reliability a proper refactoring was in order, so this ended up being a bigger rewrite than intended.

 ------

More details from commit message:
While providing an option to connect to Redis via Unix socket (#32), a
bug was introduced that made majority of the scripts silently ignore the
socket option and use default host/port instead. This was mainly due to
bad copypasta.

This commit, unifies all the CLI options in one module
(RedisClientOptions) and provides unit tests. Also, as a regression
suite, all scripts have basic tests added to make sure they properly
include the module.

This change should also make it easy to maintain a consistent UX by
having one place where options, defaults and description strings can be
modified.

### CODE REVIEW

**Comment by mbyczkowski** [comment]
@majormoses @eheydrick any thoughts on this?

**Comment by majormoses** [comment]
@mbyczkowski sorry I actually started a review several days ago but never got through it. Let me take a look.

**Comment by mbyczkowski** [comment]
Thanks @majormoses! Yeah, I noticed the integration tests have just landed recently. I'm going to rebase everything and try to rewrite tests to be integration tests. `redis_endpoint` sounds like a good method name to me 👌

**Comment by majormoses** [comment]
@mbyczkowski just checking if you are looking to get back to this. I know a lot of this stuff falls off peoples radars after holidays.

**Comment by mbyczkowski** [comment]
Yes @majormoses, it did fall off my radar in past couple weeks, but I'm still planning to do the rebase/clean-up some time in Dec. As soon as I get some time. Thanks for the ping! :)

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-redis/versions/2.3.1
